RPATH handling and installation paths fix

eessmann opened the pull request:
Hi Tyson
This code adjusts the install interface paths to use CMAKE_INSTALL_INCLUDEDIR and adds RPATH handling using the helper function setup_quest_rpath. This ensures consistent RPATH behaviour across platforms.
Erich

eessmann committed:
RPATH handling and installation paths

Adjusts install interface paths to use CMAKE_INSTALL_INCLUDEDIR
and adds RPATH handling using helper function setup_quest_rpath.
Ensures consistent RPATH behavior across platforms

eessmann committed:
Sets RPATH origin for dynamic linking

TysonRayJones commented:
Wew thanks very much for looking at this! Could you PR this instead to the [`devel`](https://github.com/QuEST-Kit/QuEST/tree/devel) branch? Then we can easily test it with changes made to the examples building, and bundle it with a `v4.1` merge into `main`.

eessmann commented:
Sure, no problem
The RPATH fix should clear up some issues I have been having on my quest rust bindings side project

eessmann committed:
Include path fix

TysonRayJones commented:
Thanks very much! Btw do you think it's worth testing _execution_ of the executables compiled during CI? The compiled examples (at least, those in `examples/isolated`) run in mere seconds, but could reveal a build issue like the bundling of the DLLs (as per #592).

If that's worthwhile, we could bundle the workflow change into this PR to test these build changes "for free". It _should_ be as simple as adding code like below to the bottom of [`compile.yml`](https://github.com/QuEST-Kit/QuEST/blob/devel/.github/workflows/compile.yml):
```yml
      # attempt to run the compiled executable, testing e.g. DLL location
      - name: Run
        if: ${{ matrix.os == 'windows-latest' }}
        shell: bash
        run: |
          ./${{ env.build_dir }}/examples/isolated/Release/complex_arithmetic_c.exe
      - name: Run
        if: ${{ matrix.os != 'windows-latest' }}
        run: |
          ./${{ env.build_dir }}/examples/isolated/complex_arithmetic_c
```

If you think it's a good idea, I'll PR to your `rpath-fix` branch and we can squash-commit this PR together. Alternatively, you can make the above above edits yourself to [`compile.yml`](https://github.com/QuEST-Kit/QuEST/blob/devel/.github/workflows/compile.yml), though that might prove a headache; it will require my manually re-triggering the CI (auto-permissions disabled until the paid runners aren't on my credit card hehe), and experiencing the general horror of tinkering with Github Actions.

eessmann committed:
Add base to installed include path

eessmann commented:
I think it's a great idea to run an isolated exe for a few secs in CI. It should catch most link time errors

TysonRayJones committed:
made compilation CI run compiled example

specifically the `complex_arithmetic_c` example. This serves just to check for link-time errors due to problems in the build

TysonRayJones committed:
redirect Windows output to log

TysonRayJones committed:
still trying to capture Windows exec output

TysonRayJones committed:
changed executed example

since 'complex_arithmetic_c' was actually disabled on Windows and so output nothing, hehe!

TysonRayJones committed:
correcting path typo

TysonRayJones committed:
updated comment

TysonRayJones committed:
temporarily disabling LCOV reporting
